lib/manager/pub/readme.md

```diff
@@ -0,0 +1,277 @@
+## Overview
+
+#### Name of package manager
+
+[pub](https://pub.dartlang.org/)
+
+---
+
+#### What language does this support?
+
+Dart / Flutter SDK
+
+---
+
+#### Does that language have other (competing?) package managers?
+
+No.
+
+## Package File Detection
+
+#### What type of package files and names does it use?
+
+_pubspec.yaml_
+
+---
+
+#### What [fileMatch](https://renovatebot.com/docs/configuration-options/#filematch) pattern(s) should be used?
+
+`^pubspec\.yaml`
+
+---
+
+#### Is it likely that many users would need to extend this pattern for custom file names?
+
+No
+
+---
+
+#### Is the fileMatch pattern likely to get many "false hits" for files that have nothing to do with package management?
+
+No
+
+## Parsing and Extraction
+
+#### Can package files have "local" links to each other that need to be resolved?
+
+Yes
+
+#### Is there reason why package files need to be parsed together (in serial) instead of independently?
+
+No
+
+---
+
+#### What format/syntax is the package file in? e.g. JSON, TOML, custom?
+
+YAML
+
+---
+
+#### How do you suggest parsing the file? Using an off-the-shelf parser, using regex, or can it be custom-parsed line by line?
+
+Use YAML parser like [JS-YAML](https://github.com/nodeca/js-yaml)
+
+---
+
+#### Does the package file structure distinguish between different "types" of dependencies? e.g. production dependencies, dev dependencies, etc?
+
+Yes, `dependencies` and `dev-dependencies`.
+
+More details in: <https://www.dartlang.org/tools/pub/dependencies>
+
+Example from <https://github.com/brianegan/flutter_architecture_samples/blob/master/example/firestore_redux/pubspec.yaml>:
+
+```yaml
+environment:
+  sdk: '>=2.0.0 <3.0.0'
+
+dependencies:
+  meta: '>=1.1.0 <2.0.0'
+  redux: ^3.0.0
+  flutter_redux: ^0.5.0
+  flutter:
+    sdk: flutter
+  flutter_architecture_samples:
+    path: ../../
+  firebase_flutter_repository:
+    path: ../firebase_flutter_repository
+
+dev_dependencies:
+  test: ^1.3.0
+  mockito: ^3.0.0
+  flutter_driver:
+    sdk: flutter
+  flutter_test:
+    sdk: flutter
+  integration_tests:
+    path: ../integration_tests
+  todos_repository_flutter:
+    path: ../todos_repository_flutter
+```
+
+---
+
+#### List all the sources/syntaxes of dependencies that can be extracted:
+
+- [**SDK**](https://www.dartlang.org/tools/pub/dependencies#sdk)
+  ```yaml
+  dependencies:
+    flutter_driver:
+      sdk: flutter
+      version: ^0.0.1
```

Comment:
From my reading, this is a very special case, i.e. flutter is the only currently supported value under `sdk`. Should Renovate upgrade its version/value by default or leave it as-is?

Reply:
I think it could be left as is. 

The most common use case is not to have a value and depend on the `Flutter SDK` version:
```yaml
dependencies:
  flutter:
    sdk: flutter

  flutter_localizations:
    sdk: flutter

  built_value: ^6.3.0
  built_collection: ^4.1.0
```